### ISSUE TITLE
Player's money doesn't work

### ISSUE THREAD

**Issue by SopelPL** [SkriptLang/Skript#2288]
### Description
Hi, I have a problem with Vault economy in my script. If i try send %player's money%, i have a error, like this "Can't understand this condition/effect". I have vault plugin, skQuery, Skellet, Skript 3.7 and skVault.

### Server Information
* **Server version/platform:**      Spigot 1.12.2
* **Skript version:**               Skript 3.7

**Comment by OfficialDonut** [comment]
Skript 3.7? that's not a version? Anyways, do you have an economy plugin? you need one for the money expression to work. skvault is for creating your own economy in skript. if youre not doing that you dont need it

**Comment by Matocolotoe** [comment]
Well the syntax you gave us is incorrect (you forgot the quotes) so in this case Skript is right

**Comment by TheLimeGlass** [comment]
You need Skript latest found https://github.com/bensku/Skript/releases
Latest Vault https://dev.bukkit.org/projects/vault
and latest Spigot for best tests.

An economy plugin is not needed, Skript acts as the economy plugin with Skript code, it's only useful when you wanna read from that economy plugin. Unrelated.

`send "%player's balance%"` is a proper syntax, you may have not inserted the quotes as needed.

It may also not be working due to outdated Skript or Vault, there was a bug in older versions of Skript where vault wouldn't hook due to old implementation.

**Comment by Blueyescat** [comment]
An economy plugin is needed for the money expression as already its docs says. It wouldn't work with SkVault since it works in runtime and Skript registers the syntaxes on startup. 

https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/hooks/VaultHook.java#L61-L62

**Comment by TheLimeGlass** [comment]
The link you provided is just the API for extending economy, there is only one economy aka Vault. You don't need an economy plugin, that's what Vault is, it allows plugins to manipulate the currency.

**Comment by ShaneBeee** [comment]
Vault requires an economy plugin to function. If there is no economy plugin (ie: essentials) then vault disables its economy stuff, therefor Skript will not recognize the player's balance, since there isn't one.

**Comment by Blueyescat** [comment]
consider investigating before replying

**Comment by TheLimeGlass** [comment]
You linked the API to extend Skript's economy bruh

**Comment by ShaneBeee** [comment]
> You linked the API to extend Skript's economy bruh

Yeah and he is showing:
```java
	protected void loadClasses() throws IOException {
		if (economy != null)
			Skript.getAddonInstance().loadClasses(getClass().getPackage().getName() + ".economy");
```
If there is no economy plugin (ie: essentials), Vault's economy will = null, and Skript's syntaxes for economy will not load

**Comment by TheLimeGlass** [comment]
Works for me, it might be bukkit permissions maybe, and not the actual vault.

**Comment by ShaneBeee** [comment]
tl;dr

Vault **is not** an economy plugin. Vault is an API link between economy plugins (such as Essentials, TheNewEconomy or even SkVault) and plugins which use economies (such as JobsReborn, Towny, Skript)

**Comment by TheLimeGlass** [comment]
I know it's an API, in which Skript extends to use within Skript.

**Comment by ShaneBeee** [comment]
> The link you provided is just the API for extending economy, there is only one supported economy aka Vault. You don't need an economy plugin, that's what Vault is,



> I know it's an API

Are you sure about that? Your comment prior you are stating that Vault is an economy plugin, when in fact, it is most definitely NOT an economy plugin

**Comment by Blueyescat** [comment]
Lime you are discussing just for self-righteous

**Comment by TheLimeGlass** [comment]
I swear it worked back in the day when I didn't make my own currency, might have been an addon? You should really stop rambling in an issue.

**Comment by Blueyescat** [comment]
It never ever worked without **an economy**, you are a Java developer so shouldn't be hard to understand these things. You are who doing that, please don't *cause* another hundreds of comments after this, all comments after Matocolotoe's one were just unnecessary.

**Comment by SopelPL** [comment]
Yes i forgot economy plugin. Thanks for all replies.

**Comment by TheLimeGlass** [comment]
No Problem, glad you solved it.
